backend/wmg/pipeline/summary_cubes/expression_summary_fmg/load.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+import numpy as np
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+def build_in_mem_cube(
+    gene_ids: pd.DataFrame,
+    cube_index: pd.DataFrame,
+    other_cube_attrs: list,
+    cube_sum: np.ndarray,
+    cube_sqsum: np.ndarray,
+    cube_nnz: np.ndarray,
+    cube_nnz_thr: np.ndarray,
+):
+    """
+    Build the cube in memory, calculating the gene expression value for each combination of attributes
```

Comment:
I see a lot of duplication between the two `build_in_mem_cube`. Have you tried factoring out reusable portions?

Reply:
I tried, but it's a bit difficult because it makes the required input arguments overly verbose. I will think about it more to see if we can get a clean abstraction. Otherwise, I don't see the harm in having similar functions living in isolated modules. It's the most straightforward for grokking the code, imo.

Reply:
As long as making a change in one function will not require to make the same change in the other.